### DISCUSSION TITLE
[admob] How can I detect a banner not showing to the user?

### DISCUSSION THREAD

**Discussion by edodusi** [invertase/react-native-firebase#4764] [Q&A]
Scenario: I set a banner with `<BannerAd>`, a unit and a size. That specific banner is not showing: every other unit is showing, but for that specific unit there is no content.

Problem: even if there is no content to show, `onAdFailedToLoad` is not called (there's no error) and `onAdLoaded` is called.

So how can I possibly detect that the banner is not visible to the user, there is no content, so that I can collapse the container or show something else - or load a different unit?

**Comment by mikehardy** [comment]
This is not immediately useful, for which I apologize, but all I can recommend is reaching right into node_modules and instrumenting the code with as much console.log (javascript) and native logging as possible in order to trace it.

**Reply by edodusi** [reply]
ok so you don't think there's a documented way to do that right from the `BannerAd` component and we need a new feature here possibly?

**Reply by mikehardy** [reply]
I can't take a position without evidence showing what the actual problem is, sorry - thus the instrumentation

**Reply by edodusi** [reply]
ok will try with logging as much as possible
